src/reducers/query.ts

```diff
@@ -87,16 +87,32 @@ export default {
   query: handleActions(
     {
       [ActionTypes.PREVIEW_DATA_SUCCESS]: (state, action: any) => {
-        const rows = action.payload.queryResults.data.result;
         const columnsByName = _.keyBy(state.columns, 'name');
-        const columns = action.payload.columns.map((column: ColumnModel) => ({
+        const columns: TableColumnType[] = action.payload.columns.map((column: ColumnModel) => ({
           name: column.name,
           dataType: column.datatype,
           arrayOf: column.array_of,
           allowSort: !column.array_of,
           allowResize: true,
           width: columnsByName[column.name]?.width || TABLE_DEFAULT_COLUMN_WIDTH,
         }));
+        // We only need to re-format row data of type timestamp
+        const timestampColumns: TableColumnType[] = [];
+        columns.forEach((col: TableColumnType) => {
+          if (col.dataType === 'timestamp') {
+            timestampColumns.push(col);
+          }
+        });
+        const rows = action.payload.queryResults.data.result.map((row: any) => {
+          timestampColumns.forEach((col: TableColumnType) => {
+            if (col.arrayOf) {
+              row[col.name] = row[col.name].map((v: number) => new Date(v * 1000).toLocaleString());
```

Comment:
What happens if v is null or 0?  I think we might need a check here no?

Reply:
Oh, great point. Yes, if it was null, I was just setting it to the Unix epoch. I also added handling for empty arrays in the LightTable code.
<img width="678" alt="image" src="https://user-images.githubusercontent.com/13254229/235735325-c288b804-5e70-4a84-bae1-3327c46305c6.png">